firstvoices/backend/serializers/membership_serializers.py

```diff
@@ -54,19 +56,37 @@ class Meta:
         fields = ("role",) + SiteSummarySerializer.Meta.fields
 
 
+class WriteableUserEmailSerializer(serializers.SlugRelatedField):
+    def to_representation(self, value):
+        return UserDetailSerializer(context=self.context).to_representation(value)
+
+
 class MembershipDetailSerializer(WritableSiteContentSerializer):
     url = SiteHyperlinkedIdentityField(
-        view_name="api:membership-detail", read_only=True
+        read_only=True, view_name="api:membership-detail"
     )
-    user = UserDetailSerializer(allow_null=False, read_only=True)
-    role = serializers.CharField(source="get_role_display")
+    user = WriteableUserEmailSerializer(
+        allow_null=False,
+        slug_field="email",
+        queryset=User.objects.all(),
+        validators=[UniqueForSite(queryset=Membership.objects.all())],
```

Comment:
Should the queryset here be Users instead of Memberships?

Reply:
Gotcha gotcha. Looking more closely, this validator only works for site content models so I think this is the best option. Thanks for checking!